pkg/kudoctl/cmd/install/install.go

```diff
@@ -201,27 +201,37 @@ func installFramework(name, previous string, repository repo.FrameworkRepository
 	// This is also the part you end up when no dependencies are found or installed and all Framework and
 	// FrameworkVersions are already installed.
 
+	// First make sure that our instance object is up to date with overrides from commandline
+	applyInstanceOverrides(crds.Instance, options, isDependencyInstall)
+
 	// Check if Instance exists in cluster
-	// It won't create the Instance if any in combination with given Framework Name and FrameworkVersion exists
-	if !kc.AnyInstanceExistsInCluster(name, options.Namespace, crds.FrameworkVersion.Spec.Version) {
+	// It won't create the Instance if any in combination with given Framework Name, FrameworkVersion and Instance frameworkName exists
+	instanceExists, err := kc.InstanceExistsInCluster(frameworkName, options.Namespace, crds.FrameworkVersion.Spec.Version, crds.Instance.ObjectMeta.Name)
+	if err != nil {
+		return errors.Wrapf(err, "verifying the instance does not already exist")
+	}
+
+	if !instanceExists {
 		// This happens when the given FrameworkVersion is not existing. E.g.
 		// when a version has been installed that is not part of the official kudobuilder/frameworks repo.
 		if !options.AutoApprove {
 			fmt.Printf("No Instance tied to this \"%s\" version has been found. "+
-				"Do you want to create one? (Yes/no) ", name)
+				"Do you want to create one? (Yes/no) ", frameworkName)
 			if helpers.AskForConfirmation() {
 				// If Instance is a dependency we need to make sure installSingleInstanceToCluster is aware of it.
 				// By having the previous string set we can make this distinction.
-				if err := installSingleInstanceToCluster(name, previous, crds.Instance, kc, options); err != nil {
+				if err := installSingleInstanceToCluster(frameworkName, crds.Instance, kc, options); err != nil {
 					return errors.Wrap(err, "installing single Instance")
 				}
 			}
 		} else {
-			if err := installSingleInstanceToCluster(name, previous, crds.Instance, kc, options); err != nil {
+			if err := installSingleInstanceToCluster(frameworkName, crds.Instance, kc, options); err != nil {
 				return errors.Wrap(err, "installing single Instance")
 			}
 		}
 
+	} else {
+		fmt.Printf("Cannot install Instance ")
```

Comment:
Can this message be more descriptive?

Reply:
lol :D it looks like I fell asleep when wrtiting this :D good catch